mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs

```diff
@@ -39,7 +39,7 @@ void InitializeForCurrentThread (int skipFrames, bool needFileInfo)
 
 			StackFrame sf;
 			var frames = new List<StackFrame> ();
-			while ((sf = new StackFrame (skipFrames, needFileInfo)) != null && sf.GetMethod () != null) {
+			while (skipFrames >= 0 && (sf = new StackFrame (skipFrames, needFileInfo)) != null && sf.GetMethod () != null) {
```

Comment:
Note that the `new StackFrame(...) != null` doesn't make sense, but it was already in the code before.

Reply:
Could you please fix that as well

Reply:
Done